src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingCrafter.java

```diff
@@ -181,6 +181,16 @@ public static boolean canBuildingCanLearnMoreRecipes(final int buildingLevel, fi
         return (Math.pow(2, buildingLevel) * EXTRA_RECIPE_MULTIPLIER) >= (learnedRecipes + 1);
     }
 
+    /**
+     * Returns the maximum number of recipes a building may have
+     * @param buildingLevel the building level
+     * @return maxinum number of recipes
+     */
+    public static int buildingMaxRecipes(final int buildingLevel)
```

Comment:
If we're going to have a central place like this, we need to update the "canAddRecipe" code to use it, otherwise we have a chance of it getting out of sync.

Reply:
thinking more about this, you probably need a static and an instance version of this code. I think the instance should be _public int getMaxRecipeCount()_

Reply:
We should never ever have a static version of this code.
Please refactor this so that it is building/worker specific.

There is too much context required to make this calculation even remotely feasible as a static method.

Reply:
Yeah, ask the building in the view sync for the max and synch it over instead of this additional static method